src/Microsoft.AspNet.Authentication/ClaimsTransformer.cs

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Security.Claims;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNet.Authentication
+{
+    public class ClaimsTransformer : IClaimsTransformer
+    {
+        public Func<ClaimsPrincipal, Task<ClaimsPrincipal>> OnTransform { get; set; }
+
+        public virtual Task<ClaimsPrincipal> TransformAsync(ClaimsPrincipal principal)
+        {
+            if (OnTransform != null)
+            {
+                return OnTransform(principal);
```

Comment:
Well won't this prevent the delegate from return a null principal?  At least before the transform could transform into null